src/test/getarg_tests.cpp

```diff
@@ -183,4 +183,36 @@ BOOST_AUTO_TEST_CASE(boolargno)
     BOOST_CHECK(gArgs.GetBoolArg("-foo", false));
 }
 
+BOOST_AUTO_TEST_CASE(logargs)
+{
+    const auto okaylog_bool = std::make_pair("-okaylog-bool", ArgsManager::ALLOW_BOOL);
+    const auto okaylog_negbool = std::make_pair("-okaylog-negbool", ArgsManager::ALLOW_BOOL);
+    const auto okaylog = std::make_pair("-okaylog", ArgsManager::ALLOW_ANY);
+    const auto dontlog = std::make_pair("-dontlog", ArgsManager::ALLOW_ANY | ArgsManager::DONT_LOG);
+    SetupArgs({okaylog_bool, okaylog_negbool, okaylog, dontlog});
+    ResetArgs("-okaylog-bool -nookaylog-negbool -okaylog=public -dontlog=private");
+
+    // Log the arguments to debug.log
+    gArgs.LogArgs();
+
+    // Open and read debug.log.
+    FILE* file = fsbridge::fopen(LogInstance().m_file_path, "rb");
```

Comment:
Should `#include <fs.h>` be explicitly added to this file? Other code and unit tests using `fsbridge` seem to include it, and according to developer-notes.md:

```md
- Every `.cpp` and `.h` file should `#include` every header file it directly uses classes, functions or other
  definitions from, even if those headers are already included indirectly through other headers.

  - *Rationale*: Excluding headers because they are already indirectly included results in compilation
    failures when those indirect dependencies change. Furthermore, it obscures what the real code
    dependencies are.
```

Reply:
Other than that, the new unit tests look good and provide helpful feedback when run with

`src/test/test_bitcoin -t getarg_tests/logargs -l all`

Reply:
I think we could test this without reading an actual file by registering a log callback with `PushBackCallback` (and `DeleteCallback` afterwards).

Reply:
I think there is also an `ASSERT_DEBUG_LOG` which might come in handy